packages/lesswrong/server/languageModels/promptUtils.ts

```diff
@@ -288,14 +319,18 @@ The postId and commentIds (the _id in each comment) are given in the search resu
 <CurrentPostComments>\n${await formatCommentsForPost(currentPost, tokenCounter, context)}\n</CurrentPostComments>\n\n`
     : '';
 
+  const providedPostsBlock = providedPosts.length > 0
+    ? `The user mentioned the following posts in their query. They are presumed to be EXTREMELY RELEVANT to answering the users query. Other posts might be relevant too, but these are the ones the user mentioned first, so they are likely very important: <UserProvidedPosts>\n${formatAdditionalPostsForPrompt(providedPosts, tokenCounter)}\n</UserProvidedPosts>\n\n`
```

Comment:
I think we should strive quite hard to avoid typos (and grammatical mistakes, etc) in prompts to LLMs, i.e. `users query` should be `user's query`.

Reply:
(I'm also worried about what happens in the case where `formatAdditionalPostsForPrompt` returns an empty string because of the token counter; might be worth precomputing that result first:

```
let providedPostsBlock = '';
if (providedPosts.length > 0) {
  const formattedAdditionalPostsContent = formatAdditionalPostsForPrompt(...);
  if (formattedAdditionalPostsContent.length > 0) {
    providedPostsBlock = `...`;
  }
}
```

or something (you'll want to check the inner length condition actually does the thing it's supposed to do)

Reply:
The complexity-to-hastily-written ness of the token counting logic here is admittedly higher than for the rest of the code I've written.

Reply:
;)

<img width="550" alt="promptUtils ts — LessWrong2 2024-10-25 18-03-31" src="https://github.com/user-attachments/assets/cf7abb9d-4c27-402c-98ab-db971024c48c">